### DISCUSSION TITLE
Decode AVIF error: File does not support the 'heic' brand.

### DISCUSSION THREAD

**Discussion by entGriff** [libvips/libvips#2294] [Q&A] [answered]
Hello, 

I need some image manipulations on AVIF format and  using `libvips` via `ubuntu:20.04` docker container. 

Have  installed necessary ubuntu libraries and also  passing `--with-heif \` option to `./configure`. 

Here are libheif encoders and decoders:
```
# pkg-config libheif --print-variables
builtin_h265_decoder
builtin_h265_encoder
exec_prefix
includedir
libdir
pcfiledir
prefix
```

The results from `./configure`:
```
...
file import with libheif:               yes
...
```

Heifload also  looks good:
```
# vips heifload

load a HEIF image
usage:
   heifload filename out [--option-name option-value ...]
where:
   filename     - Filename to load from, input gchararray
   out          - Output image, output VipsImage
optional arguments:
   page         - Load this page from the file, input gint
...
```

**My problem  is that when I trying to convery `avif` file to any other format, getting following error:**
```
# vips copy images/avif-orientation-6.avif IMG_0344.jpg
images/avif-orientation-6.avif: bad seek to 162019
heif: Unsupported file-type: Unspecified: File does not support the 'heic' brand.
 (3.0)

```
This particular `avif` file is from `libvips/test/images`, but the same happens  with any other AVIF files. 

On the other hand, it works when I converting  for example jpg to avif.

btw, some of the tests failes when I running `make check`:
```
FAIL: test_connections.sh
PASS: test_descriptors.sh
FAIL: test_cli.sh
FAIL: test_formats.sh
PASS: test_seq.sh
PASS: test_stall.sh
PASS: test_threading.sh

```

Any suggestions or hints will be really  helpful. 

Thanks!

**Comment by jcupitt** [comment, marked as answer]
Hi @ent1c3d,

I think you might be missing the AVIF libraries. I see:

```
$ pkg-config libheif --print-variables
builtin_avif_decoder
builtin_avif_encoder
builtin_h265_decoder
builtin_h265_encoder
exec_prefix
includedir
libdir
pcfiledir
prefix
```

It's confusing: libheif supports both HEIC and AVIF, and each needs an appropriate backend enabled.

This could also be causing the test suite failures.

**Reply by entGriff** [reply]
Thanks for your quick response @jcupitt ! 

Can you please suggest AVIF libraries for ubuntu 20.04?

**Reply by jcupitt** [reply]
I think most people suggest dav1d and rav1e, but of course there's a lot of choice.

**Reply by entGriff** [reply]
@jcupitt  Just looking most convenient one to be bundled lib and to install it via `apt-get` easily, to avoid some tricky installs, but can't find any so far. 

Also, do I need  to link the backend with libheif somehow, for example with `./configure`?

**Reply by jcupitt** [reply]
Looks like I'm using libaom-dev and libdav1d-dev. You'll need to reconfigure libheif after installing the encoders and encoders.

**Reply by kleisauke** [reply]
fwiw, libheif can also be installed from strukturag's PPA (which includes AVIF support via libaom), this is what we do in our CI:
https://github.com/libvips/libvips/blob/06b429955a2528f8e36ae0eef3f6fbf4e4a95922/.github/workflows/ci.yml#L41-L42

**Reply by entGriff** [reply]
@jcupitt @kleisauke 
Thank you so much both of you!

Your comments were really  helpful and it started working. Thanks!
